### PR TITLE
Fixes the server taking for fucking ever to initialize for local test

### PR DESCRIPTION
**Pull request by MrsTonedOne**
Initializations complete within 334 seconds!
vs
Initializations complete within 177.1 seconds!

(on my slow ass laptop)

Stupid edge case, stop_lag sleeps for extra time if world.cpu is over 100, but default config doesn't call stop_lag until 500% of the way into the tick during mc init, meaning world.cpu would be creeping up to 500%. Basically this was all being silly and stupid and sleeping for long periods of time to try to make up for the server being overworked, and it only makes sense to do this during the round, not during pre-game.

### CODE REVIEW

**Comment by MrStonedOne** [comment]
old
```
Loaded station in 31.5s!
Initialized Mapping subsystem within 47.5 seconds!
Initialized 815609 atoms
Initialized Atoms subsystem within 62.5 seconds!
Initialized Language subsystem within 0 seconds!
Initialized Machines subsystem within 0.3 seconds!
Initialized Shuttle subsystem within 40.9 seconds!
Initialized Weather subsystem within 0 seconds!
Initialized Atmospherics subsystem within 124.1 seconds!
Initialized Minimap subsystem within 0 seconds!
Initialized Assets subsystem within 0 seconds!
Initialized Icon Smoothing subsystem within 11.6 seconds!
Initialized Overlay subsystem within 18.5 seconds!
Initialized XKeyScore subsystem within 0 seconds!
Initialized Lighting subsystem within 27.5 seconds!
Initialized Persistence subsystem within 0 seconds!
Initializations complete within 334 seconds!
```

**Comment by MrStonedOne** [comment]
new

```
Loaded station in 17.8s!
Initialized Mapping subsystem within 27.9 seconds!
Initialized Atoms subsystem within 32.3 seconds!
Initialized Language subsystem within 0 seconds!
Initialized Machines subsystem within 0.2 seconds!
Initialized Shuttle subsystem within 20 seconds!
Initialized Weather subsystem within 0 seconds!
Initialized Atmospherics subsystem within 68.1 seconds!
Initialized Minimap subsystem within 0 seconds!
Initialized Assets subsystem within 0 seconds!
Initialized Icon Smoothing subsystem within 6.2 seconds!
Initialized Overlay subsystem within 8.9 seconds!
Initialized XKeyScore subsystem within 0 seconds!
Initialized Lighting subsystem within 13.5 seconds!
Initialized Persistence subsystem within 0 seconds!
Initializations complete within 177.1 seconds!
```

**Comment by JJRcop** [comment]
old
```
Loaded station in 13.2s!
Initialized Mapping subsystem within 20.6 seconds!
Initialized 813944 atoms
Initialized Atoms subsystem within 24.3 seconds!
Initialized Language subsystem within 0 seconds!
Initialized Machines subsystem within 0.1 seconds!
Initialized Shuttle subsystem within 15.2 seconds!
Initialized Weather subsystem within 0 seconds!
Initialized Atmospherics subsystem within 45.1 seconds!
Initialized Minimap subsystem within 0 seconds!
Initialized Assets subsystem within 0 seconds!
Initialized Icon Smoothing subsystem within 4.9 seconds!
Initialized Overlay subsystem within 7.2 seconds!
Initialized XKeyScore subsystem within 0 seconds!
Initialized Lighting subsystem within 10.4 seconds!
Loaded 0 engraved messages on map Box Station
Initialized Persistence subsystem within 0 seconds!
Initializations complete within 128.4 seconds!

```

new
```
Loaded station in 7.5s!
Initialized Mapping subsystem within 12.4 seconds!
Initialized 817063 atoms
Initialized Atoms subsystem within 15.2 seconds!
Initialized Language subsystem within 0 seconds!
Initialized Machines subsystem within 0.1 seconds!
Initialized Shuttle subsystem within 7.9 seconds!
Initialized Weather subsystem within 0 seconds!
Initialized Atmospherics subsystem within 26.4 seconds!
Initialized Minimap subsystem within 0 seconds!
Initialized Assets subsystem within 0 seconds!
Initialized Icon Smoothing subsystem within 2.7 seconds!
Initialized Overlay subsystem within 4.4 seconds!
Initialized XKeyScore subsystem within 0 seconds!
Initialized Lighting subsystem within 5.7 seconds!
Loaded 0 engraved messages on map Box Station
Initialized Persistence subsystem within 0.1 seconds!
Initializations complete within 75 seconds!

```

**Comment by lzimann** [comment]
On my vm:
old:
```
Loaded station in 5.6s!
Initialized Mapping subsystem within 13 seconds!
Initialized 813907 atoms
Initialized Atoms subsystem within 30.7 seconds!
Initialized Language subsystem within 0 seconds!
Initialized Machines subsystem within 0.2 seconds!
Initialized Shuttle subsystem within 23.6 seconds!
Initialized Weather subsystem within 0 seconds!
Initialized Atmospherics subsystem within 94.6 seconds!
Initialized Minimap subsystem within 0.1 seconds!
Initialized Assets subsystem within 0 seconds!
Initialized Icon Smoothing subsystem within 6.2 seconds!
Initialized Overlay subsystem within 15.1 seconds!
Initialized XKeyScore subsystem within 0 seconds!
Initialized Lighting subsystem within 13.2 seconds!
Initialized Persistence subsystem within 0.5 seconds!
Initializations complete within 198.4 seconds!
```
new: 
```
Loaded station in 3.1s!
Initialized Mapping subsystem within 8.2 seconds!
Initialized 816450 atoms
Initialized Atoms subsystem within 17.1 seconds!
Initialized Language subsystem within 0 seconds!
Initialized Machines subsystem within 0.2 seconds!
Initialized Shuttle subsystem within 11.8 seconds!
Initialized Weather subsystem within 0 seconds!
Initialized Atmospherics subsystem within 51.5 seconds!
Initialized Minimap subsystem within 0.1 seconds!
Initialized Assets subsystem within 0 seconds!
Initialized Icon Smoothing subsystem within 3.5 seconds!
Initialized Overlay subsystem within 12.3 seconds!
Initialized XKeyScore subsystem within 0 seconds!
Initialized Lighting subsystem within 7.6 seconds!
Initialized Persistence subsystem within 1.8 seconds!
Initializations complete within 114.3 seconds!
```

Quite the improvement.

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -1210,6 +1210,9 @@ proc/pick_closest_path(value, list/matches = get_fancy_list_of_atom_types())
 #define DELTA_CALC max(((max(world.tick_usage, world.cpu) / 100) * max(Master.sleep_delta,1)), 1)
 
 /proc/stoplag()
+	if (!Master || !(Master.current_runlevel & RUNLEVELS_DEFAULT))
```
Comment (Cyberboss) [inline code/__HELPERS/unsorted.dm]: RUNLEVELS_DEFAULT is setup, ingame, and post game. Doesn't count lobby. Maybe you want just !Master.current_runlevel? because if initializations are done it will always be non-zero

Comment (MrStonedOne) [inline code/__HELPERS/unsorted.dm]: No, This works better because its living.

The loop sleep system was designed to prevent contention among alot of things wanting to do shit at once, because sleep has a fairly costly overhead, so sleeping, waking up, only to sleep again because the tick is already gone, is wasteful.

This edge case is generally only likely to occur when subsystems are running, so tying this to the same define that controls when most subsystems run makes senses.

**Comment by MrStonedOne** [comment]
@tgstation/commit-access every coder, mapper, and spriter will thank you to merge this

**Comment by Cyberboss** [comment]
Yeah, let me get home

**Comment by lzimann** [comment]
I was just waiting on the 24h.
